src/components/page-details.jsx

```diff
@@ -43,12 +43,14 @@ export default class PageDetails extends React.Component {
 
   componentWillUnmount () {
     window.removeEventListener('keydown', this);
+    document.title = 'Scanner';
   }
 
   /**
    * @param {PageDetailsProps} previousProps
    */
   componentDidUpdate (previousProps) {
+    document.title = `Scanner | ${this.state.page.url}`;
```

Comment:
As a side note, you’re getting `UnhandledPromiseRejectionWarning` stuck in front of the actual error message because we are failing to handle errors triggered by the API in `_loadPage()`, but that’s not something you necessarily need to worry about fixing in this PR.